blueman/main/DNSServerProvider.py

```diff
@@ -1,19 +1,79 @@
 import re
-from ipaddress import IPv4Address
+import socket
+from ipaddress import IPv4Address, ip_address
 from typing import List, Generator
 
-from gi.repository import GObject
+from gi.repository import GObject, Gio, GLib
+
+from blueman.bluemantyping import GSignals
 
 
 class DNSServerProvider(GObject.GObject):
     RESOLVER_PATH = "/etc/resolv.conf"
 
+    __gsignals__: GSignals = {"changed": (GObject.SignalFlags.NO_HOOKS, None, ())}
+
     def __init__(self) -> None:
         super().__init__()
+        self._subscribe_systemd_resolved()
+        self._subscribe_resolver()
 
     @classmethod
     def get_servers(cls) -> List[IPv4Address]:
-        return list(cls._get_servers_from_resolver())
+        return list(set(cls._get_servers_from_systemd_resolved()) or cls._get_servers_from_resolver())
+
+    @staticmethod
+    def _get_servers_from_systemd_resolved() -> Generator[IPv4Address, None, None]:
```

Comment:
Wouldn't a Gio.DBusProxy on `org.freedesktop.resolve1.Manager` make things a lot easier here?

Reply:
Would it? Sugar for `org.freedesktop.DBus.Properties.Get` is only in `blueman.bluez.Base.Base` and all three calls in this method have distinct (object path, interface)-tuples, so that they'd need different proxy objects.

Reply:
The call method on the dbusproxy takes an optional interface name,  like `org.freedesktop.resolve1.Manager.GetLink`. as long as you are dealing with the same dbus object/name you should be able to access any interface like this.

Reply:
Example to get the DNS property but this works on anything on `/org/freedesktop/resolve1`

```python
from gi.repository import GLib, Gio
proxy = Gio.DBusProxy.new_for_bus_sync(
    Gio.BusType.SYSTEM,
    Gio.DBusProxyFlags.DO_NOT_AUTO_START,
    None,
    "org.freedesktop.resolve1",
    "/org/freedesktop/resolve1",
    "org.freedesktop.resolve1"
    )

proxy.call_sync("org.freedesktop.DBus.Properties.Get", GLib.Variant('(ss)', ("org.freedesktop.resolve1.Manager", "DNS")), Gio.DBusCallFlags.NONE, -1, None)
```
which gives 
```python
GLib.Variant('(v)', (<[(3, 2, [byte 0xc0, 0xa8, 0x01, 0x01])]>,))
```

Reply:
Ok, here is why I suggest using proxies.

```python
from gi.repository import GLib, Gio

def on_properties_changed(proxy, changed, invalidated):
    changed_properties = changed.unpack()
    if "DNS" in changed_properties:
        print(changed["DNS"])

manager_proxy = Gio.DBusProxy.new_for_bus_sync(
    Gio.BusType.SYSTEM,
    Gio.DBusProxyFlags.DO_NOT_AUTO_START,
    None,
    "org.freedesktop.resolve1",
    "/org/freedesktop/resolve1",
    "org.freedesktop.resolve1.Manager"
    )


manager_proxy.connect("g-properties-changed", on_properties_changed)
dns = manager_proxy.get_cached_property("DNS")
link_path = manager_proxy.GetLink("(i)", 2)

link_proxy = Gio.DBusProxy.new_for_bus_sync(
    Gio.BusType.SYSTEM,
    Gio.DBusProxyFlags.DO_NOT_AUTO_START,
    None,
    "org.freedesktop.resolve1",
    link_path,
    "org.freedesktop.resolve1.Link"
    )

default = link_proxy.get_cached_property("DefaultRoute")
```

Reply:
I think I did not get that last comment and I'm unsure where we're at with this PR. :sweat_smile: Do you suggest additional changes?